modules/internal/ChapelRange.chpl

```diff
@@ -610,19 +610,32 @@ module ChapelRange {
   }
 
   /* Return the range's low bound. If the range does not have a low
-     bound the behavior is undefined. */
-  inline proc range.low  return chpl_intToIdx(_low);
+     bound (e.g., ``..10``), a compiler error is generated. */
+  inline proc range.low {
+    if !hasLowBound() {
+      compilerError("can't query the low bound of a range without one");
+    }
+    return chpl_intToIdx(_low);
+  }
 
 
   /* Return the range's high bound. If the range does not have a high
-     bound the behavior is undefined. */
-  inline proc range.high return chpl_intToIdx(_high);
+     bound (e.g., ``1..``), a compiler error is generated. */
+  inline proc range.high {
+    if !hasHighBound() {
+      compilerError("can't query the high bound of a range without one");
+    }
+    return chpl_intToIdx(_high);
+  }
 
 
   /* Returns the range's aligned low bound. If the aligned low bound is
-     undefined (does not exist), the behavior is undefined.
+     undefined (e.g., ``..10 by -2``), the behavior is undefined.
```

Comment:
So a range that has a low bound can still exhibit undefined behavior here?

Reply:
Uh, no...  It seems my short attention span got the better of me (both in writing it and in reading the rendered result).  Hang on...